src/main/java/de/tum/in/www1/artemis/service/StudentExamService.java

```diff
@@ -248,6 +252,43 @@ else if (exercise instanceof ModelingExercise) {
         }
     }
 
+    /**
+     * Automatically assess the modeling- and programming exercises of student exams of an exam which are not submitted with 0 points.
+     * The assessment is counted as {@link AssessmentType#SEMI_AUTOMATIC} to make sure it is not considered for manual assessment,
+     * see {@link StudentParticipationRepository#findByExerciseIdWithLatestSubmissionWithoutManualResultsAndNoTestRunParticipation}.
+     * The assessor is current instructor who made the request.
+     *
+     * @param examId the exam id
+     * @return returns the number of assessedSubmissions
+     */
+    public int automaticallyAssessUnsubmittedExams(final Long examId) {
+        int numberOfAssessedSubmissions = 0;
+        User instructor = userService.getUser();
+        Set<StudentExam> unsubmittedStudentExams = findAllUnsubmittedStudentExams(examId);
+        Map<User, List<Exercise>> exercisesOfUser = unsubmittedStudentExams.stream().collect(Collectors.toMap(StudentExam::getUser, studentExam -> studentExam.getExercises()
+                .stream().filter(exercise -> exercise instanceof ModelingExercise || exercise instanceof TextExercise).collect(Collectors.toList())));
+        for (final var user : exercisesOfUser.keySet()) {
+            final var studentParticipations = participationService.findByStudentIdAndIndividualExercisesWithEagerSubmissionsResult(user.getId(), exercisesOfUser.get(user));
+            for (final var studentParticipation : studentParticipations) {
+                if (studentParticipation.findLatestSubmission().isPresent() && studentParticipation.findLatestSubmission().get().getLatestResult() == null) {
+                    // get last submission
+                    final var latestSubmission = studentParticipation.findLatestSubmission().get();
+                    // create result with 0 points
+                    Result result = new Result();
+                    result.setParticipation(studentParticipation);
+                    result.setAssessor(instructor);
+                    result.setCompletionDate(ZonedDateTime.now());
+                    result.setScore(0L);
+                    result.rated(true);
+                    result.setAssessmentType(AssessmentType.SEMI_AUTOMATIC);
```

Comment:
Anything going against 
```suggestion
                    result.setAssessmentType(AssessmentType.AUTOMATIC);
```
?

Reply:
We specifically exclude `SEMI_AUTOMATIC` and `MANUAL` assessments from eligible submissions for new manual assessment. If we mark it as `AUTOMATIC`, it would appear to the tutors for assessment.

Reply:
Wait, why?
I'd assume that `AssessmentType.AUTOMATIC` would represent results that have not been seen by a human. E.g. Quizzes and most programming results.
`MANUAL` is only assessed by humans and `SEMI_AUTOMATIC` is that there is an automatic assessment already there (e.g. programming tests, Compass, or Athene) which is reviewed and possibly adjusted by humans.

As the empty submissions are not supposed to be seen by a human at all, I'd argue that it should be in fact `AUTOMATIC`. I'd think of such a submission not showing up for assessment in the dashboard as it is already assessed before. (Compare with another tutor, only that it is a bot in this case.)

Reply:
@jpbernius As we discussed in private I set the assessor to be the `artemis_admin` user and left the assessment type currently to SEMI_AUTOMATIC. Maybe we can have a look at the assessment types and their usages in another PR to unify their meaning for different exercise types and decouple the rated functionality from them.

Reply:
I think we can live with the choice of SEMI_AUTOMATIC here. But please add very detailed comments into code why this decision was made and why AUTOMATIC is not possible here. It is very important that future developers immediately understand this when looking at the code